python/ray/dag/dag_node_operation.py

```diff
@@ -148,8 +148,12 @@ def tie_breaker(lhs: "_DAGOperationGraphNode", rhs: "_DAGOperationGraphNode"):
             if is_nccl_write and other_is_nccl_write:
                 return tie_breaker(self, other)
             elif is_nccl_compute and other_is_nccl_compute:
-                return tie_breaker(self, other)
+                if self.task_idx != other.task_idx:
```

Comment:
Revert this and add TODO comment. Comment out test.

Reply:
done